src/GafferScene/Encapsulate.cpp

```diff
@@ -95,9 +95,23 @@ void Encapsulate::affects( const Plug *input, AffectedPlugsContainer &outputs )
 	}
 }
 
+IECore::PathMatcher::Result Encapsulate::filterValueChecked( const Gaffer::Context *context ) const
+{
+	IECore::PathMatcher::Result f = filterValue( context );
+	if( f & IECore::PathMatcher::AncestorMatch && !(f & IECore::PathMatcher::ExactMatch ) )
```

Comment:
Hmm, you're right.  I arrived at this because I had originally added protection just to the `else` branches in `hashObject` etc to stop default accesses to the input scene, and then rearranged the logic to this ... you're right that it actually makes no sense to test ExactMatch here.

It does kind of highlight that all we're actually doing is preventing using this code for prototype encapsulation once we get to that in the Instancer ( I know that the instanceGroup encapsulation comes first, but we are already thinking about it ).  I suppose we can address this by ripping all the actual logic out of the compute/hash methods of Encapsulate into static functions that can be called by Instancer?

Reply:
> I suppose we can address this by ripping all the actual logic out of the compute/hash methods of Encapsulate into static functions that can be called by Instancer?

I don't think we need all of the logic of the Encapsulate node though. We just need the equivalent of `hashObject()/computeObject()`, which is practically nothing.

Reply:
Oh, maybe you're right.  I was thinking we would need the set logic as well, but thinking about it , I guess we don't.